temporian/core/graph.py

```diff
@@ -112,31 +141,64 @@ def p2(title, dictionary):
                 s += f"\t{k}:{v}\n"
             s += "\n"
 
-        p2("Inputs", self.inputs)
-        p2("Output", self.outputs)
+        def p3(title, l):
+            nonlocal s
+            s += f"{title} ({len(l)}):\n"
+            for v in l:
+                s += f"\t{v}\n"
+            s += "\n"
+
+        if self.named_inputs is not None:
+            p2("Named inputs", self.named_inputs)
+        else:
+            p3("Inputs", self.inputs)
+
+        if self.named_outputs is not None:
+            p2("Named output", self.named_outputs)
+        else:
+            p3("Output", self.outputs)
         return s
 
 
-def infer_graph(
-    inputs: Optional[Dict[str, NodeInputArg]],
-    outputs: Dict[str, Node],
-) -> Tuple[Graph, Dict[str, Node]]:
-    """Extracts all the objects between the output and input nodes.
+def infer_graph_named_nodes(
+    inputs: Optional[NamedNodes], outputs: NamedNodes
+) -> Graph:
+    """Extracts the nodes in between the output and input nodes.
+
+    Unlike infer_graph, infer_graph_named_nodes requires for the input and
+    output nodes to be named.
+    """
+
+    normalized_inputs: Optional[Dict[str, Node]] = None
+    input_nodes = None
+    if inputs is not None:
+        normalized_inputs = normalize_named_nodes(inputs)
+        input_nodes = set(normalized_inputs.values())
+
+    normalized_outputs = normalize_named_nodes(outputs)
+    output_nodes = set(normalized_outputs.values())
+
+    g = infer_graph(inputs=input_nodes, outputs=output_nodes)
+    if normalized_inputs is None:
+        normalized_inputs = normalize_named_nodes(list(g.inputs))
+
+    g.set_input_node_names(normalized_inputs)
+    g.set_output_node_names(normalized_outputs)
+    return g
 
-    Fails if any inputs are missing.
+
+def infer_graph(inputs: Optional[Set[Node]], outputs: Set[Node]) -> Graph:
+    """Extracts the nodes in between the output and input nodes.
+
+    If inputs is set, fails if outputs cannot be computed from "inputs".
```

Comment:
docs: use code commas ` when referring to the function's params too for them to be rendered as code